packages/store-sync/src/query-cache/queryRECS.ts

```diff
@@ -0,0 +1,118 @@
+import { ZustandStore } from "../zustand";
+import { AllTables, QueryCondition, TableSubject } from "./common";
+import { SchemaToPrimitives, StoreConfig, Table, Tables } from "@latticexyz/store";
+import { query } from "./query";
+
+export enum QueryFragmentType {
+  Has,
+  HasValue,
+  Not,
+  NotValue,
+}
+
+type HasQueryFragment<T extends Table> = {
+  type: QueryFragmentType.Has;
+  table: T;
+};
+
+type NotQueryFragment<T extends Table> = {
+  type: QueryFragmentType.Not;
+  table: T;
+};
+
+type HasValueQueryFragment<T extends Table> = {
+  type: QueryFragmentType.HasValue;
+  table: T;
+  value: SchemaToPrimitives<T["valueSchema"]>;
+};
+
+type NotValueQueryFragment<T extends Table> = {
+  type: QueryFragmentType.NotValue;
+  table: T;
+  value: SchemaToPrimitives<T["valueSchema"]>;
+};
+
+export function Has<T extends Table>(table: T): HasQueryFragment<T> {
```

Comment:
might be nice to think about how we can support customizing the "subject" (defaulting to primary key)

if you have a composite key on one table and you want to combine with another and normalize their "subjects" across the query (so that the query is valid/possible)

Reply:
If I understand, I'm not sure how to do this in RECS land - the assumption is that you operate over the entire key tuple so you can't specify subjects more finely